src/App.jsx

```diff
@@ -14,45 +17,130 @@ export const goodsFromServer = [
+  const [listOfGoods, setListOfGoods] = useState([...goodsFromServer]);
+  const [targetInnerText, setTargetInnerText] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+
+  const sortList = (list, type) => {
+    switch (type) {
+      case 'Sort alphabetically':
+        if (isReverse) {
+          return list.sort((good1, good2) => good2.localeCompare(good1));
+        }
+
+        return list.sort((good1, good2) => good1.localeCompare(good2));
+
+      case 'Sort by length':
+        if (isReverse) {
+          return list.sort((good1, good2) => good2.length - good1.length);
+        }
+
+        return list.sort((good1, good2) => good1.length - good2.length);
+
+      case 'Reverse':
+        return list.reverse();
+
+      default:
+        return list;
+    }
+  };
+
+  const handleSort = (e) => {
+    const text = e.currentTarget.innerText;
+
+    setTargetInnerText(text);
+    setListOfGoods(sortList(listOfGoods, text));
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handlleReverse = (e) => {
+    const text = e.currentTarget.innerText;
+
+    if (isReverse) {
+      setIsReverse(false);
+    } else {
+      setIsReverse(true);
+    }
+
+    setListOfGoods(sortList(listOfGoods, text));
+  };
+
+  const handleReset = () => {
+    setListOfGoods(sortList([...goodsFromServer], ''));
+    setTargetInnerText('');
+    setIsReverse(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            cn(
+              'button is-info',
+              { 'is-light': targetInnerText !== 'Sort alphabetically' },
+            )
+          }
+          onClick={(e) => {
+            handleSort(e);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn(
+              'button is-success',
+              { 'is-light': targetInnerText !== 'Sort by length' },
+            )
+          }
+          onClick={(e) => {
+            handleSort(e);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn(
+              'button is-warning',
+              { 'is-light': !isReverse },
+            )
+          }
+          onClick={(e) => {
+            handlleReverse(e);
+          }}
+        >
+          Reverse
+        </button>
+
+        {
+          (targetInnerText || isReverse)
+            ? (
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={() => {
+                  handleReset();
+                }}
```

Comment:
when your handler accepts an event as arg or doesn't accept any you can just write:
```suggestion
                onClick={handleReset}
```